src/cia.rs

```diff
@@ -176,3 +176,59 @@ impl GameController {
         (position, fire)
     }
 }
+
+bitflags! {
+    #[derive(Clone, Copy)]
+    pub struct VicBankSelect: u8 {
+        const VIC_C000 = 0b0000_0000;
+        const VIC_8000 = 0b0000_0001;
+        const VIC_4000 = 0b0000_0010;
+        const VIC_0000 = 0b0000_0011;
+   }
+}
+
+bitflags! {
+    #[derive(Clone, Copy)]
+    pub struct SerialBusAccess: u8 {
+        const TXD_LO   = 0b0000_0100;
+        const ATNO_LO  = 0b0000_1000;
+        const CLKO_LO  = 0b0001_0000;
+        const DTAO_LO  = 0b0010_0000;
+        const CLKO_HI  = 0b0100_0000;
+        const DATI_HI  = 0b1000_0000;
+    }
+}
+
+bitflags! {
+    #[derive(Clone, Copy)]
+    pub struct RS232Access: u8 {
+    }
+}
+
+#[repr(C, packed)]
+pub struct MOSComplexInterfaceAdapter6526_2 {
```

Comment:
Instead of duplicating the struct, use generics, i.e.
```rust
pub struct MOSComplexInterfaceAdapter6526<T1: Copy, T2: Copy> {
    pub port_a: RW<T1>,                // 0x00
    pub port_b: RW<T2>,                // 0x01
```
Then set `CIA1` and `CIA2` to the appropriate flags.